### PR TITLE
Add 12 inplace APIs including auto generated

### PR DESCRIPTION
**Pull request by pangyoki**
### PR types

New features

### PR changes

APIs

### Describe

Inplace Strategy has been implemented in PR #30103 .
Add 12 Inplace APIs in this PR.

### NOTICE
The newly added inplace APIs in this PR will not be used publicly. They can only be called in the form of `Tensor.api_`, but CANNOT be called in the form of `paddle.api_`.
- There is no corresponding `Tensor.api` method for activation op, such as `relu`, `hard_sigmoid`, `leaky_relu`, `sigmoid` and  , so its inplace method will not be added temporarily.
- The input of `add_n` API is a Tensor list, and it is not convenient to use `Tensor.api` to call, so its inplace method will not be added temporarily. 

### Processed API Lists in this PR

|Inplace OP | paddle下API | 
|---|---|
| clip | paddle.tensor.math.clip | 
| scale | paddle.tensor.math.scale |  
| elementwise_sub | paddle.tensor.math.subtract |  
| elementwise_add | paddle.tensor.math.add | 

Add 8 Inplace APIs that are auto generated.

|Inplace OP |  paddle下API | 
|---|---|
| ceil |  paddle.tensor.math.ceil | 
| floor | paddle.tensor.math.floor | 
| exp | paddle.tensor.math.exp | 
| reciprocal |  paddle.tensor.math.reciprocal | 
| round |  paddle.tensor.math.round | 
| sqrt |  paddle.tensor.math.sqrt | 
| rsqrt | paddle.tensor.math.rsqrt | 

PS: `square` grad op needs to use the input var, so the corresponding Inplace API `square_` hasn't been added.

Fix `flatten_contiguous_range` op kernel, and add `flatten_` Inplace API.
|Inplace OP |  paddle下API | 
|---|---|
| flatten_contiguous_range | paddle.tensor.manipulation.flatten | 

### 17 Inplace APIs in Tensor method

```
Tensor.reshape_
Tensor.squeeze_
Tensor.unsqueeze_
Tensor.scatter_
Tensor.tanh_
Tensor.clip_
Tensor.scale_
Tensor.add_
Tensor.subtract_
Tensor.ceil_
Tensor.floor_
Tensor.exp_
Tensor.reciprocal_
Tensor.round_
Tensor.sqrt_
Tensor.rsqrt_
Tensor.flatten_
```

### example to use Inplace APIs

```
import paddle

a = paddle.rand([2,3])
# [[0.45528218, 0.92556411, 0.90491837],
#   [0.31372914, 0.01990727, 0.94381261]]

b = paddle.ones([2,3])
# [[1., 1., 1.],
# [1., 1., 1.]]

a.add_(b)
print(a)
# [[1.45528221, 1.92556405, 1.90491843],
#  [1.31372917, 1.01990724, 1.94381261]]

paddle.add_(a,b)
# Traceback (most recent call last):
#   File "<stdin>", line 1, in <module>
# AttributeError: module 'paddle' has no attribute 'add_'

```

### CODE REVIEW

**Review by zhiqiu — DISMISSED** [review]
LGTM

**Review by lanxianghit — DISMISSED** [review]

File: python/paddle/nn/functional/loss.py [hunk 1]
```diff
@@ -37,8 +37,32 @@
...
 from ...fluid.framework import _varbase_creator
 from ...fluid.framework import Variable
+from ...fluid.dygraph.inplace_utils import inplace_apis_in_dygraph_only
 from paddle.utils import deprecated
 
+kIgnoreIndex = -100
+
+
+@inplace_apis_in_dygraph_only
+def softmax_with_cross_entropy_(logits,
```
Comment (jzhang533) [inline python/paddle/nn/functional/loss.py]: softmax_with_cross_entropy API 已经被标记为deprecated了，所以这个inplace版本应该是也不需要了。

**Review by jzhang533 — COMMENTED** [review]
LGTM for tools/wlist.json now, but we do have plan to gradually cleanup this white list.

File: python/paddle/nn/functional/loss.py (hunk 1 above)
Comment (pangyoki) [reply python/paddle/nn/functional/loss.py]: 已经删除了inplace版本的softmax_with_cross_entropy_，谢谢

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
LGTM

**Pull request merged**